### DISCUSSION TITLE
Add email login alternative: a delegator Newbie Villa contract

### DISCUSSION THREAD

**Discussion by Atlasoin** [Crossbell-Box/Crossbell-Contracts#7] [Ideas]
To bridge more traditional Web2 creators to Crossbell, a more convenient(especially mobile-friendly) approach is important. Inspired by discussions with lots of people, here I propose a design for Newbie Villa:

## Register
1. Alternative to connecting with a wallet, newcomers could register via email address and then mint a new character. Each email address is then bound to this character.
2.  All new characters will be created by an admin and transferred to the Newbie Villa contract. All interactions from that character will be forwarded to the Newbie Villa contract. And all gas will temporarily(at least at the current stage) be paid by the admin.
3. The total number of interactions should be limited. It's reasonable to limit the gas used corresponding with the faucet limit.

## Withdraw
1. Either newcomers are comfortable with the crypto wallet system, or be motivated by the aggregated assets, they might choose to withdraw their character from the Newbie Villa to their designated address.
2. Default withdrawal method would be: login with the email and connect with the designated wallet address(with enough $CSB to pay gas), then sign the withdrawal tx.
3. Beside the default way, I strongly propose an "offline" withdrawal verification approach: when the character is created, a proof will be generated and encrypted, and sent to the registered email box. Later that proof could be used to interact with Newbie Villa's contract to withdraw the character directly. 

For the entire flow and two different withdrawal approaches, I also drafted a simple
[illustration](https://whimsical.com/newbie-villa-98zp1He5NXS7LiqsR2YfGC).

**Comment by songkeys** [comment]
> 3. The total number of interactions should be limited. It’s reasonable to limit the gas used corresponding with the faucet limit.

Why should it be limited? Is it to urge them to have a wallet or for the huge amount of $CSB spent or? If the former, why would we do that if the user is happy about the email login method? If the latter, I don't think that's a big issue for us cuz currently the validators are all our admins'.

**Reply by Atlasoin** [reply]
E.g 0.02 $CSB already supports huge amounts of transactions. That would not cause "obvious" inconvenience. I just feel the limitation should correspond to the faucet limitation, to make a logically sound economic system.

**Reply by songkeys** [reply]
> supports huge amounts of transactions

Not that huge. Different types of tx vary. E.g.,

- a (first) [linkNote](https://scan.crossbell.io/tx/0x374625c541cef791d9f00424f4f9b8ba3ed06de3b65e70a768b3d00c21f7d8d8) tx cost me ~0.0001 CSB. That is, one could only (first) like at most 200 notes, without any other txs.
- a (first) [mintNote](https://scan.crossbell.io/tx/0x88445e0d47482e69e4da32e25729c1d1d95d00a2fcff8fb64ae6962685912b13) tx cost me ~0.0003 CSB. That is up to ~66 mints.
- a (non-first) [linkCharacter](https://scan.crossbell.io/tx/0xd1d4f1f2bb4bb81ba2e95d3a082251521b045198e1676977b46f3d159d8f876f) tx cost me ~0.0001 CSB. That is up to ~200 following.

> the limitation should correspond to the faucet limitation

But wallet user could withdraw from the faucet once running out of $CSB while email user couldn't. This is the big issue. They have to migrate to wallet to continue. **If this is not what we want**, we should provide some way to prevent this.

> to make a logically sound economic system.

Looks like this is the reasoning behind. Then, a possible alternative comes to my mind: no limitation; user can use email as a login proof for whatever long, with a **$CSB incentive model** - 

- **Starting Point**: When registered, user will be granted 0.02 $CSB to interact.
- **Cycling Period**: A daily/monthly/long-term task system is introduced. When accomplishing a task (e.g., post a note today), user will be able to click “receive” button (manually) for some amount of $CSB, so as to keep interacting in a healthy model.

The same model may apply to users using wallets to connect - the faucet is the starting point and the task system is the cycling period.

The benefits:

- **No more email->wallet migration happens**.
- **Healthy and reasonable incentive model for users to interact.** The more you interact (as a proof of an active user), the more you should get $CSB to interact.
- **A possible gate for $CSB to be meaningful.** Currently $CSB plays a boring role as gas fee. To earn it, users have to withdraw from the faucet, the only place. With the model proposed above, it could be made into a reasonable currency in the future because we have an internal circulation within the app (i.e., `spend <==> earn`).

**Reply by Atlasoin** [reply]
Good point! 
> user will be able to click “receive” button (manually) for some amount of $CSB

I love this point. By doing that, even email users could have an explicit feeling of the existence of $CSB and the behind the economic system.

And for $CSB incentive model, I do love this idea! But it's very challenging to build a healthy and reasonable incentive model without falling into a complete "fi"-driven community. Proof of a real human is not that easy, especially if we don't want to introduce those disgusting user-tracking plugins. I believe we need to discuss more and maybe also do some small experiments.

> No more email->wallet migration happens

Honestly I am not sure if that's a benefit or not. It's true that in the economic sense, there are no "migration". But what's worth persuading is each one owns his data without relying upon the trust for others. Email users still rely on the trust of the admin of Newbie Villa or the login server. So the ideal solution I could see is to help users aggregate enough assets on Crossbell to motivate themselves to keep their assets.

So to make a temporary conclusion, how about integrating a button in the app to get $CSB from the faucet(with the same rules limited to wallet users)? As for a better $CSB incentive model, currently, we could discuss more before we make a final decision.

**Reply by songkeys** [reply]
> > No more email->wallet migration happens
> 
> Honestly I am not sure if that's a benefit or not. It's true that in the economic sense, there are no "migration". But what's worth persuading is each one owns his data without relying upon the trust for others. Email users still rely on the trust of the admin of Newbie Villa or the login server. So the ideal solution I could see is to help users aggregate enough assets on Crossbell to motivate themselves to keep their assets.

I should say "No more *forcible* email->wallet migration happens". Those who care will do it; the rest stays in their comfort zoom as they would love to. The only thing we should do is preaching but not forcing from which a possible loss a group of users.

> how about integrating a button in the app to get $CSB from the faucet(with the same rules limited to wallet users)? As for a better $CSB incentive model, currently, we could discuss more before we make a final decision.

I do agree this needs more thinking and discussion, and a some time to develop. Integrating a "faucet" button looks like a first low-fidelity implementation of task system, which could be called "daily or weekly check-in task". Let's discuss this further within the next team meeting.
